### ISSUE TITLE
Unable to install on Ubuntu 24.04 LTS

### ISSUE THREAD

**Issue by oparada1988** [davidchalifoux/keylight-commander#25]
Pretty much tittle, 

![Screenshot from 2024-08-20 11-47-41](https://github.com/user-attachments/assets/d25a8f47-69a5-4294-b3f5-0af4993de31d)

searching for libwebkit2gtk in the repository gives me:

![Screenshot from 2024-08-20 11-48-49](https://github.com/user-attachments/assets/9c69f369-b046-4eec-9678-d8f053464e88)

Which is already installed:

![Screenshot from 2024-08-20 11-49-14](https://github.com/user-attachments/assets/e439ff79-537b-42f3-ad0f-22e7822e25dc)

**Comment by davidchalifoux** [comment]
Strange, I'll try to reproduce this. Does it work if you install with `dpkg -i`?
In the meantime you can use the AppImage version.

**Comment by oparada1988** [comment]
nope :-( 

![Screenshot from 2024-08-20 14-00-20](https://github.com/user-attachments/assets/f63af092-5b63-4fc5-984c-eaf9b2995e71)

**Comment by Lawouach** [comment]
Hi there,

Just for what it's worth, I have a different error on Ubuntu 24.10:

```
./keylight-commander_0.0.12_amd64.AppImage  -h
/usr/lib/x86_64-linux-gnu/gvfs/libgvfscommon.so: undefined symbol: g_task_set_static_name
Failed to load module: /usr/lib/x86_64-linux-gnu/gio/modules/libgvfsdbus.so

** (keylight-commander:61409): WARNING **: 13:36:02.516: webkit_settings_set_enable_offline_web_application_cache is deprecated and does nothing.
/usr/lib/x86_64-linux-gnu/gvfs/libgvfscommon.so: undefined symbol: g_task_set_static_name
Failed to load module: /usr/lib/x86_64-linux-gnu/gio/modules/libgvfsdbus.so
STARTED
/usr/lib/x86_64-linux-gnu/gvfs/libgvfscommon.so: undefined symbol: g_task_set_static_name
Failed to load module: /usr/lib/x86_64-linux-gnu/gio/modules/libgvfsdbus.so

```

Using the AppImage

**Comment by Lawouach** [comment]
I have the same dependency error on libwebkit2gtk-4.0-37 when using the deb package.

**Comment by davidchalifoux** [comment]
I'm starting a rework of this project with Tauri v2 that will fix this issue: https://github.com/tauri-apps/tauri/issues/9662

Will update when a new version is available.

**Comment by davidchalifoux** [comment]
I've been able to make a lot of progress on the rewrite. I'm currently aiming for a release this weekend.

This new version will be significantly more efficient from a state management perspective. Some small things like setting the name on the hardware level will be added too. Stay tuned!

**Comment by oparada1988** [comment]
Thank you for your update, looking foward to it! Are you considering a flatpak release?

**Comment by davidchalifoux** [comment]
> Thank you for your update, looking foward to it! Are you considering a flatpak release?

Yeah that'd be a good idea, I'll aim for adding more deploy targets in general to make installing easier.

**Comment by davidchalifoux** [comment]
The v0.1.0 pre-release is available now: https://github.com/davidchalifoux/keylight-commander/releases/tag/v0.1.0

Please give it a try and let me know if you run into any issues! I haven't had the chance to test on Windows/Linux yet. I'll add more targets (Flathub/etc) once I can confirm this is a good version.

If you are running a version earlier than this it will not automatically update when this is released. You'll have to manually re-install.

**Comment by oparada1988** [comment]
Ubuntu 24.04.2 LTS works great, only "issue" I see is you cant drag the window if you leave "hide on focus change" you have to disable it if you want to move it. 

Also, it would be cool if the windos could remember its position after closing

**Comment by Lawouach** [comment]
Hi @davidchalifoux thank you for your work!

On Ubuntu 24.10 with my Elgato Key Air, it works fine so that's fantastic. I've used the AppImage.

I only noticed that the app seems to disappear when clicking outside it. It doesn't seem to stop, the UI just goes away and it doesn't show up on the dock anywhere so I can't show it again.

Also, a few messages on startup:

/usr/lib/x86_64-linux-gnu/gvfs/libgvfscommon.so: undefined symbol: g_task_set_static_name
Failed to load module: /usr/lib/x86_64-linux-gnu/gio/modules/libgvfsdbus.so
/usr/lib/x86_64-linux-gnu/gvfs/libgvfscommon.so: undefined symbol: g_task_set_static_name
Failed to load module: /usr/lib/x86_64-linux-gnu/gio/modules/libgvfsdbus.so
/usr/lib/x86_64-linux-gnu/gvfs/libgvfscommon.so: undefined symbol: g_task_set_static_name
Failed to load module: /usr/lib/x86_64-linux-gnu/gio/modules/libgvfsdbus.so
Created service daemon

A couple of features would be neat:

* a close button on the UI
* that it remembers the light and it doesn't ask to Search for one everytime
* a preset with a suitable default for common use cases

These are only nice to have, as it is I can at least use the app to drive the light and that's awesome :)

**Comment by oparada1988** [comment]
On bazzite the appimg doesnt even open, I tried opening it from the terminal but no errors

https://github.com/user-attachments/assets/f1d7d4e3-b668-4363-9696-c591bbd467c3

**Comment by davidchalifoux** [comment]
Just updated the pre-release version, grab the latest download: https://github.com/davidchalifoux/keylight-commander/releases/tag/v0.1.0

**What's new:**
- The tray icon and menu were re-implemented on the Rust side. Should fix issues where the icon wasn't showing before.
- There's a new tray menu option for showing the app while keeping it above any other apps, should help with some funky Gnome focus timers.
- Fixed scanning on Windows, it'll now use the IPv4 instead of the bonjour hostname.
- Fixed an issue that could cause scanning to continuously retry.


> On bazzite the appimg doesnt even open, I tried opening it from the terminal but no errors

What desktop environment are you using?


> Ubuntu 24.04.2 LTS works great, only "issue" I see is you cant drag the window if you leave "hide on focus change" you have to disable it if you want to move it.

I'm not getting this behavior on Ubuntu 24.10, but it is an known issue on [Windows](https://github.com/tauri-apps/tauri/issues/10767). Could you share a screen recording?


> Also, it would be cool if the windos could remember its position after closing

This should be fixed now, at least when hiding and showing the app.


> I only noticed that the app seems to disappear when clicking outside it. It doesn't seem to stop, the UI just goes away and it doesn't show up on the dock anywhere so I can't show it again.

It should show up on your system tray, from there you can hide and show it. This is meant to be more of a "tray" app so it won't be visible on your task bar.

There is a setting for disabling the automatic hiding though!


> A couple of features would be neat:
> 
> * a close button on the UI
> * that it remembers the light and it doesn't ask to Search for one everytime
> * a preset with a suitable default for common use cases

- You can close from the tray icon menu. If that ends up too annoying I could add buttons for it on the window too.
- Right now it resets the list of lights on restart, but not on hide/show of the app. I figure this is fine enough but I can chache it if needed. Let me know if this is still annoying after using the tray icon to keep it running.
- Presets seem unnecessary IMO since it's fast to change them on-the-fly. If more people want this I can add it.

**Comment by davidchalifoux** [comment]
> Ubuntu 24.04.2 LTS works great, only "issue" I see is you cant drag the window if you leave "hide on focus change" you have to disable it if you want to move it.
> 
> Also, it would be cool if the windos could remember its position after closing

Interestingly I wasn't getting this behavior with the AppImage, but I am with the deb package. I think my best solution for now will be to turn decorations back on so you have the native window management bars.

**Comment by oparada1988** [comment]
> I tested the latest release on all platforms, looks good to me. I re-enabled the native window controls so you shouldn't run into the auto-hide issue.
> 
> I _did_ see some weird behavior when closing and re-opening the window on the deb build, but not the AppImage build. For now I recommend sticking to the AppImage on Linux.
> 
> [@oparada1988](https://github.com/oparada1988) is the latest AppImage working on Bazzite for you?

Unfortunately it only gives me a blank windows
